packages/frontend/src/hooks/useEcharts.ts

```diff
@@ -30,16 +32,12 @@ import { getDimensionFormatter } from '../utils/resultFormatter';
 import { useOrganisation } from './organisation/useOrganisation';
 
 const getLabelFromField = (
-    explore: Explore,
-    tableCalculations: TableCalculation[],
+    fields: Array<Field | TableCalculation>,
     key: string | undefined,
 ) => {
-    const field = key ? findFieldByIdInExplore(explore, key) : undefined;
-    const tableCalculation = tableCalculations.find(({ name }) => name === key);
+    const field = findItem(fields, key);
     if (field) {
-        return getFieldLabel(field);
-    } else if (tableCalculation) {
-        return tableCalculation.displayName;
+        return isField(field) ? getFieldLabel(field) : field.displayName;
```

Comment:
did you remove the tableCaclulation here ? that was added to fix a bug on labels

Reply:
No, table calculation is still being handled. The util function returns a dimension or metric or table calculation then in this line we check if is a dimension/metric ( `isField(field)` ). I should rename the var `field` with `item` since it brings confusion.